invenio/modules/accounts/forms.py

```diff
@@ -31,13 +31,83 @@
 
 from wtforms.fields import BooleanField, HiddenField, PasswordField, \
     StringField, SubmitField
-from wtforms.validators import DataRequired
+from wtforms.validators import DataRequired, StopValidation, ValidationError
 
 from .models import User
 from .validators import validate_email, validate_nickname, \
     validate_nickname_or_email, wash_login_method
 
 
+def nickname_validator(form, field):
+    """Validate nickname."""
+    validate_nickname(field.data)
+    # is nickname already taken?
+    try:
```

Comment:
I know that Python is "Easier to ask for forgiveness than permission."

But I must admit that it is a bit tricky to read quickly.  

My 2cents but I think it will be better to do an 

except NoResultFound ( perhaps even MultipleResultsFound, even if in theory it should never happen)
Because there we will caught all the exception fro SQLAlchemy even the one that could be terribly annoying. 
Wouldn't it be a good idea to register the exception in this case ? (with a second except with SQLAlchemyError ?)